### PR TITLE
refactor(node): follow-ups to deterministic_ips/ consolidation

### PR DESCRIPTION
**Pull request by andrewbattat**
NODE-1533

- Move Deployment into types.rs: 16f80c19268140d03f0a2511d1f5047f25ec5e5a
- Add node_type.rs (taken from the old mac_address library): 9d672af52632176b8786c7cf2792105ad42fb0bb
- Add macaddr crate: 554f6bbf9b2ad2c2efa7962a57f9bca302c8fa2c
- Assorted deterministic_ips refactoring (d5ce1163ead0c6462e47f61f66bcc8985c3e85dd, eac304a28977ad6221b4d489b6d4e3b44cbe3ce8)
- Remove HwAddr type wrapper and use MacAddr6 directly: 7bb6ee9d0a6f538e615cbc36c39a817e0d7ffcd8

### CODE REVIEW

**Comment by andrewbattat** [comment]
Hourly pipeline: https://github.com/dfinity/ic/actions/runs/12057813655

File: rs/ic_os/deterministic_ips/src/lib.rs [hunk 1]
```diff
@@ -1,30 +1,61 @@
...
-    fn octets(&self) -> [u8; 6] {
-        [self.a, self.b, self.c, self.d, self.e, self.f]
+    pub fn octets(&self) -> [u8; 6] {
+        self.0
+            .as_bytes()
+            .try_into()
+            .expect("MAC address should always be 6 bytes")
+    }
+    /// Generates the SLAAC IPv6 address for the given prefix.
+    pub fn calculate_slaac(&self, prefix: &str) -> Result<Ipv6Addr> {
```
Comment (frankdavid) [inline rs/ic_os/deterministic_ips/src/lib.rs:23]: Could you check that the existing tests provide enough coverage?

**Review by frankdavid — APPROVED** [review]

File: rs/ic_os/deterministic_ips/src/lib.rs (hunk 1 above)
Comment (andrewbattat) [reply rs/ic_os/deterministic_ips/src/lib.rs:24]: I just added a few more unit tests 👍

File: rs/ic_os/deterministic_ips/src/main.rs [hunk 2]
```diff
@@ -1,79 +0,0 @@
...
-    index: u8,
-) -> anyhow::Result<Ipv6Addr> {
-    // For now, this tool only outputs IPv6
-    let mac = calculate_deterministic_mac(mac, deployment, IpVariant::V6, index)?;
-    let ip = mac.calculate_slaac(prefix)?;
-
-    Ok(ip)
-}
-
-fn main() -> anyhow::Result<()> {
```
[the remark this answers is not in the document]
Comment (andrewbattat) [reply rs/ic_os/deterministic_ips/src/main.rs]: Ah, I didn't know it was ever used! Added back in.

**Review by Bownairo — APPROVED** [review]
Looks great!

**Review by marko-k0 — APPROVED** [review]

**Pull request merged**